src/pyk/kcfg/explore.py

```diff
@@ -78,6 +82,7 @@ def __init__(
     ):
         self.kprint = kprint
         self.id = id if id is not None else 'NO ID'
+        self.semantics = semantics
```

Comment:
```suggestion
        self.semantics = semantics or DefaultSemantics()
```

If the type checker complains:

```suggestion
        self.semantics = semantics if semantics is not None else DefaultSemantics()
```